src/charm.py

```diff
@@ -98,9 +128,15 @@ def _on_config_changed(self, event):
         """Handle changed configuration."""
         self.publish_config()
 
-    def publish_config(self):
+    def _on_server_cert_changed(self, _):
+        self._update_cert()
+        self._on_nginx_pebble_ready(_)
+        self.publish_config(tls=self._is_cert_available)
```

Comment:
MimirCoordinator take a server cert in its constructor so do we need here something like
```python
self.coordinator.update_server_cert(...)
```

Reply:
For now I would keep this here, and see how it evolves.
I like the idea of moving this to `MimirCoordinator`, but on the other hand I like to have everything related to certs in the "same" place.